renderdoc/core/intervals.h

```diff
@@ -156,6 +156,27 @@ class IntervalsIter
   inline Interval *operator->() { return &ref; }
 };
 
+template <typename T>
+struct HalfInterval
+{
+  uint64_t start;
+  T value;
+  static const size_t packed_size = sizeof(uint64_t) + sizeof(T);
+};
+
+template <typename T>
+inline const char *TypeName<HalfInterval<T>>()
+{
+  return "HalfInterval<T>";
+}
```

Comment:
I would prefer not to have templated structs at the serialise level like this. I think it's fine to have the `Intervals<>` class templated since it will come in handy for other types, but in this case I'd rather have a concrete type used and serialised for a `FrameRefType` half-interval.